src/app/list/basic-list/example/list-polling-example.component.ts

```diff
@@ -0,0 +1,194 @@
+  ViewEncapsulation
+} from '@angular/core';
+
+import { cloneDeep } from 'lodash';
+
+import { Action } from '../../../action/action';
+import { ActionConfig } from '../../../action/action-config';
+import { EmptyStateConfig } from '../../../empty-state/empty-state-config';
+import { ListEvent } from '../../list-event';
+import { ListConfig } from '../list-config';
+
+@Component({
+  encapsulation: ViewEncapsulation.None,
+  selector: 'list-polling-example',
+  templateUrl: './list-polling-example.component.html'
+})
+export class ListPollingExampleComponent implements OnInit {
+  actionsText: string = '';
+  allItems: any[];
+  emptyStateConfig: EmptyStateConfig;
+  items: any[];
+  itemsAvailable: boolean = true;
+  listConfig: ListConfig;
+  actionConfig: ActionConfig;
+  selectType: string = 'checkbox';
+  updateItemsInterval: number;
+
+  constructor() {
+  }
+
+  ngOnInit(): void {
+    this.allItems = [
+      this.makeRandomItem(),
+      this.makeRandomItem(),
+      this.makeRandomItem(),
+    ];
+    this.allItems[0].expanded = true;
+    this.items = cloneDeep(this.allItems);
+
+    this.emptyStateConfig = {
+      actions: {
+        primaryActions: [{
+          id: 'action1',
+          title: 'Main Action',
+          tooltip: 'Start the server'
+        }],
+        moreActions: [{
+          id: 'action2',
+          title: 'Secondary Action 1',
+          tooltip: 'Do the first thing'
+        }, {
+          id: 'action3',
+          title: 'Secondary Action 2',
+          tooltip: 'Do something else'
+        }, {
+          id: 'action4',
+          title: 'Secondary Action 3',
+          tooltip: 'Do something special'
+        }]
+      } as ActionConfig,
+      iconStyleClass: 'pficon-warning-triangle-o',
+      title: 'No Items Available',
+      info: 'This is the Empty State component. The goal of a empty state pattern is to provide a good first ' +
+        'impression that helps users to achieve their goals. It should be used when a list is empty because no ' +
+        'objects exists and you want to guide the user to perform specific actions.',
+      helpLink: {
+        hypertext: 'List example',
+        text: 'For more information please see the',
+        url: '#/list'
+      }
+    } as EmptyStateConfig;
+
+    this.actionConfig = {
+      primaryActions: [],
+      moreActions: [{
+        id: 'hint',
+        title: 'This menu should stay open while the list updates',
+        tooltip: 'This menu should stay open while the list updates'
+      }],
+      moreActionsDisabled: false,
+      moreActionsVisible: true
+    } as ActionConfig;
+
+    this.listConfig = {
+      dblClick: false,
+      emptyStateConfig: this.emptyStateConfig,
+      multiSelect: false,
+      selectItems: false,
+      selectionMatchProp: 'name',
+      showCheckbox: true,
+      showRadioButton: false,
+      useExpandItems: true
+    } as ListConfig;
+
+    this.updateItemsInterval = setInterval(() => this.updateItems(), 2500);
```

Comment:
I'm seeing an error in my IDE, which states "Type 'Timer' is not assignable to type 'number'. However, using this seems to work:

`updateItemsInterval: number = <any>setInterval(() => {});`

See https://github.com/TypeStrong/atom-typescript/issues/1053

Reply:
I had it too but assumed it was something specific to my IDE (WebStorm). I'll update this to use `any` then.